### ISSUE TITLE
Error when embedding a timestamp in Db2 XMLATTRIBUTES

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12201]
Running this statement here:

```sql
select xmlelement(name "a", xmlattributes(timestamp '2000-03-15 10:15:30.0' as "b"))
from sysibm.dual;
```

Should produce:

```xml
<a b="2000-03-15T10:15:00"/>
```

But it fails in Db2, with:

> SQL Error [22011]: The statement was not executed because a numeric argument of a scalar function is out of range.. SQLCODE=-138, SQLSTATE=22011, DRIVER=4.28.11

A workaround would be this:

```sql
-- To produce the XML timestamp literal, which is valid ISO 8601
select xmlelement(name "a", 
  xmlattributes(replace(to_char(timestamp '2000-03-15 10:15:30.0', 'YYYY-MM-DD HH24:MI:SS.FF'), ' ', 'T') as "b"))
from sysibm.dual;
```

This is the same problem as https://github.com/jOOQ/jOOQ/issues/12160, but that issue was backported to the now-released 3.15.1 and 3.14.13 already, so I'm creating a new issue

**Comment by lukaseder** [comment]
Though, the fix is a bit more tricky than for `XMLELEMENTS()`, because we're re-using `Field.as(alias)` to provide key/value pairs in attributes...

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0, 3.15.2 (https://github.com/jOOQ/jOOQ/issues/12202), and 3.14.14 (https://github.com/jOOQ/jOOQ/issues/12203)
